### PR TITLE
Bump golang version to 1.22

### PR DESCRIPTION
**Pull request by lahsivjar**
OTel has bumped up the minimum go version to 1.22. The PR bumps up the golang version for all components to 1.22.0 and checks from 1.21.10 to 1.22.7.

This should also fix vulnerability reports like https://pkg.go.dev/vuln/GO-2024-2687 which are reported in version up PRs like https://github.com/elastic/opentelemetry-collector-components/pull/116 -- it happens because with `GOTOOLCHAIN=auto` it downloads the toolchain based on to go version in go module IF the latest local version is less than the go version.

I am doing go version update of modules here instead of dependabot because dependabot seems to like introducing the `gotoolchain` directive everywhere.

### CODE REVIEW

File: .github/workflows/checks.yml
Comment (lahsivjar) [reply .github/workflows/checks.yml:22]: Why do you think we need to do that? The version here is the one which gets used to install the go version for running checks however, in go.mod is the lowest supported version. If we keep them in sync then we would need to bump up the lowest supported version which might not be the best way forward with users using the binary...

Comment (lahsivjar) [reply .github/workflows/checks.yml:22]: > ah I see, then wouldn't it be better to use 1.21 or stable and let the action pick the latest version ?

Hmm, that's a good point. The checks seem to be heavily inspired by upstream ([ref](https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-collector-contrib+%22go-version%3A%22&type=code)) and I think the reason for not pinning the version is to ensure that the binary is checked at least till a specific go-version -- this is just my interpretation of why the upstream is not pinning the version and I may be wrong here.

**Comment by lahsivjar** [comment]
Blocked by: https://github.com/elastic/opentelemetry-collector-components/pull/121

File: .github/workflows/checks.yml
Comment (rogercoll) [reply .github/workflows/checks.yml:22]: I don't have a strong opinion on this topic, but it seems upstream pins the version to control security vulnerabilities too. e.g. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35059

**Review by rogercoll — APPROVED** [review]

File: processor/elasticinframetricsprocessor/go.mod
Comment (lahsivjar) [inline processor/elasticinframetricsprocessor/go.mod:3]: @ishleenk17 Can you review this bit? The inframetrics processor changes require approval from an appropriate team member.

**Review by ishleenk17 — APPROVED** [review]

**Pull request merged**
